src/Smalot/PdfParser/Element/ElementDate.php

```diff
@@ -59,29 +59,25 @@ class ElementDate extends ElementString
     protected $format = 'c';
 
     /**
-     * @param \DateTime $value
+     * @var \DateTime
      */
+    protected $value;
+
     public function __construct($value)
     {
         if (!($value instanceof \DateTime)) {
-            throw new \Exception('DateTime required.');
+            throw new \Exception('DateTime required.'); // FIXME: Sometimes strings are passed to this function
```

Comment:
Can you explain what you mean with the `FIXME` here?

Reply:
Yes. I could not type the input parameter `$value` there to be of type `DateTime` as would be logical, because it seemed sometimes strings were passed in there. I did not dig too much into it, but for example, in `ElementDateTest` there is a test method `testGetContentInvalidParameter`, which does exactly that. When the argument is typed, there is an error instead of an exception in such cases.
Seems that this logic should be refactored somehow. That we should always pass a `DateTime` object into the constructor. If strings are needed, there can be some `createFromString` static method available. An exception for checking the input parameter type does not seem good, but I don't know enough about how it is actually used to change it right now.